stack/stack.go

```diff
@@ -43,11 +49,26 @@ func ParseYAML(yamlFile string) (*Services, error) {
 		return nil, fmt.Errorf("'%s' is the only valid provider for this tool - found: %s", providerName, services.Provider.Name)
 	}
 
+	if len(filter) > 0 {
+		for k, function := range services.Functions {
+			function.Name = k
+			match, err := regexp.MatchString(filter, function.Name)
+			if err != nil {
+				fmt.Printf("Error parsing regex passed in through --filter argument: %v", err)
+				return nil, err
+			}
+			if !match {
+				delete(services.Functions, function.Name)
+			}
+		}
+	}
+
+	err = nil
 	return &services, err
 }
 
-// fetchYAML pulls in file from remote location such as GitHub raw file-view
-func fetchYAML(address *url.URL) ([]byte, error) {
+// fetchYaml pulls in file from remote location such as GitHub raw file-view
+func fetchYaml(address *url.URL) ([]byte, error) {
```

Comment:
should be fetchYAML (in comments and fn calls etc)

Reply:
blah, missed this one. Good catch